RyuSocks.Generator/Packet/AccessorGenerator.Array.cs

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) RyuSOCKS
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+using RyuSocks.Generator.Builder;
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace RyuSocks.Generator.Packet
+{
+    [SuppressMessage("ReSharper", "RedundantIfElseBlock")]
+    internal static partial class AccessorGenerator
+    {
+        private static string[] GenerateArrayByteAccessor(PacketFieldModel packetField, bool isGetter)
+        {
+            BlockBuilder source = new();
+
+            if (isGetter)
+            {
+                string fieldTypeName = packetField.FieldType.Name.Extract(0, -2);
+                string maybeCast = packetField.FieldType.IsEnum ? $"({fieldTypeName})" : string.Empty;
+
+                source.AppendLine($"{fieldTypeName}[] result = new {fieldTypeName}[{packetField.GetLength()}];");
+                source.AppendLine();
+                source.EnterScope($"for (int i = 0; i < {packetField.GetLength()}; i++)");
+                source.AppendLine($"result[i] = {maybeCast}this[{packetField.GetOffset()} + i];");
+                source.LeaveScope();
+                source.AppendLine();
+                source.AppendLine("return result;");
```

Comment:
Add newlines to improve discernability between normal added lines and scope
```suggestion
                source.AppendLine($"{fieldTypeName}[] result = new {fieldTypeName}[{packetField.GetLength()}];");
                source.AppendLine();
                
                source.EnterScope($"for (int i = 0; i < {packetField.GetLength()}; i++)");
                source.AppendLine($"result[i] = {maybeCast}this[{packetField.GetOffset()} + i];");
                source.LeaveScope();
                
                source.AppendLine();
                source.AppendLine("return result;");
```